jbmc/src/java_bytecode/remove_instanceof.cpp

```diff
@@ -98,11 +116,37 @@ bool remove_instanceoft::lower_instanceof(
   INVARIANT(
     target_type.id() == ID_struct_tag,
     "instanceof second operand should have a simple type");
-  const irep_idt &target_name =
-    to_struct_tag_type(target_type).get_identifier();
-  std::vector<irep_idt> children=
-    class_hierarchy.get_children_trans(target_name);
-  children.push_back(target_name);
+
+  if(
+    to_struct_tag_type(target_type).get_identifier() ==
+    "java::java.lang.Object")
+  {
+    // Everything derives from Object, and special-casing this particular check
+    // means we don't need to check all array dimensionalities, which also
+    // derive from Object, but finding the highest-dimensioned array in the
+    // program under test is not trivial.
+    expr = true_exprt();
+    return true;
+  }
+
+  // If checking for an array type, the target type must have the same
+  // dimension:
+  std::size_t array_dimensions = 0;
```

Comment:
:bulb: Pull out a utiltity for getting dimension of an array type? Maybe there already is one?

Reply:
Factored out into `java_array_dimension_and_element_type` in `java_types.h`. Recommend test-gen's `dim_and_element_type` (which is less general) should go away / use it.

Reply:
:angel:  it'd be great if you fix this in your bump!